libsolidity/ast/AST.cpp

```diff
@@ -243,6 +243,41 @@ vector<EventDefinition const*> const ContractDefinition::usedInterfaceEvents() c
 	);
 }
 
+vector<EventDefinition const*> ContractDefinition::interfaceEvents(bool _requireCallGraph) const
+{
+	set<EventDefinition const*, CompareByID> result;
+	for (ContractDefinition const* contract: annotation().linearizedBaseContracts)
+		result += contract->events();
+	solAssert(annotation().creationCallGraph.set() == annotation().deployedCallGraph.set());
+	if (_requireCallGraph)
+		solAssert(annotation().creationCallGraph.set());
+	if (annotation().creationCallGraph.set())
+		result += usedInterfaceEvents();
+	// We could filter out all events that do not have an external interface
+	// if _requireCallGraph is false.
+	return util::convertContainer<vector<EventDefinition const*>>(std::move(result));
+}
+
+vector<EventDefinition const*> ContractDefinition::natspecInterfaceEvents() const
+{
+	auto eventSignature =
+		[](EventDefinition const* _event) -> string {
+			FunctionType const* functionType = _event->functionType(true);
+			solAssert(functionType, "");
+			return functionType->externalSignature();
+	};
+	auto compareBySignature =
+		[&eventSignature](EventDefinition const* _lhs, EventDefinition const* _rhs) -> bool {
+			return eventSignature(_lhs) < eventSignature(_rhs);
+	};
```

Comment:
```suggestion
	auto compareBySignature = [&](EventDefinition const* _lhs, EventDefinition const* _rhs) -> bool {
			return eventSignature(_lhs) < eventSignature(_rhs);
	};
```
No need to explicitly capture variables by references - just using `[&]` is guaranteed to only capture the references you use within the lambda.

Reply:
It's not incorrect though and has the nice effect that you won't use anything you shouldn't be using by mistake when you don't pay attention. Especially easy to fall into when you have parameters with names close to local variables, like `_contract` and `contract`, than mean different things (though you can argue that it's just bad naming and I'd agree :)). So I always considered it a better option by default, with `&` being a good enough compromise when you're lazy or the list would be too long.

In this particular case I think I'd agree with you though. The line is long and we'd be fine with `&` here. I considered suggesting it myself, but I'm also perfectly fine with leaving it as is.

Reply:
My intention was to be careful and super explicit that the only thing this really needs from the outer context is `eventSignature`. But I agree that it could be less verbose without it and have the same effect.